rapid7/convection
# Show when resource will be replaced during a convection diff

# Description




This change affects the "convection diff" output when Amazon wants to "update" a resource using CloudFormation in a way that requires [Replacement](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement).

# Motivation and Context

The problem with updates requiring replacement is that you have to "just know" what property changes require an update.  Now, when a qualifying property change will produce a create/destroy action, the diff will show "replace" in red.

# Usage Examples

This example, from this repo, shows how changing a VPC Cidr will cause a converge to replace the VPC:
![screen shot 2016-11-08 at 4 11 53 pm](https://cloud.githubusercontent.com/assets/22451912/20119769/71ea64b4-a5cf-11e6-98b4-37cc6a4596df.png)

# Testing Steps

Before converging any changes you would like to make (or, after making a few that require replacement and a few that require updates just to see what happens) run "convection diff".  The --vv shown in the example is optional
